src/psyclone/psyir/nodes/node.py

```diff
@@ -95,9 +95,145 @@ def colored(text, _):
         return text
 
 
+class ChildrenList(list):
```

Comment:
Please explain this v nice approach and what it requires when adding new nodes in the developers guide

Reply:
Added an explanation in the developers guide.